corehq/apps/receiverwrapper/util.py

```diff
@@ -209,17 +211,23 @@ def get_app_and_build_ids(domain, build_or_app_id):
     return build_or_app_id, None
 
 
-def from_demo_user(form_json):
+def from_demo_user(form_json, domain):
     """
     Whether the form is submitted by demo_user
     """
     from corehq.apps.users.util import DEMO_USER_ID
     try:
-        # require new-style meta/userID (reject Meta/chw_id)
-        if form_json['meta']['userID'] == DEMO_USER_ID:
-            return True
+        user_id = form_json['meta']['userID']
     except (KeyError, ValueError):
         return False
+    else:
+        # require new-style meta/userID (reject Meta/chw_id)
```

Comment:
comment is no longer relevant and should be removed

Reply:
https://github.com/dimagi/commcare-hq/pull/25615/commits/711aa0be5f5189ab1237f842b2aa473ffbf2f729